modules/tensor_mechanics/src/interfacekernels/CZMInterfaceKernelIncremental.C

```diff
@@ -0,0 +1,168 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CZMInterfaceKernelIncremental.h"
+
+registerMooseObject("TensorMechanicsApp", CZMInterfaceKernelIncremental);
+
+InputParameters
+CZMInterfaceKernelIncremental::validParams()
+{
+  InputParameters params = InterfaceKernel::validParams();
+  params.addRequiredParam<unsigned int>("component",
+                                        "The component of the "
+                                        "displacement vector this kernel is working on:"
+                                        " component == 0, ==> X"
+                                        " component == 1, ==> Y"
+                                        " component == 2, ==> Z");
+  params.set<bool>("_use_undisplaced_reference_points") = true;
+
+  params.addRequiredCoupledVar("displacements", "The string containing displacement variables");
+
+  params.addClassDescription(
+      "Interface kernel for use with incremental cohesive zone models (CZMs) "
+      "computing the traction as a function of the displacement jump. This kernel supports large "
+      "deformations by setting large_kinematic=true in the associated cohesive material.");
+
+  return params;
+}
+
+CZMInterfaceKernelIncremental::CZMInterfaceKernelIncremental(const InputParameters & parameters)
+  : InterfaceKernel(parameters),
+    _component(getParam<unsigned int>("component")),
+    _ndisp(coupledComponents("displacements")),
+    _disp_var(_ndisp),
+    _disp_neighbor_var(_ndisp),
+    _vars(_ndisp),
+    _PK1traction(getMaterialPropertyByName<RealVectorValue>("PK1traction")),
+    _dPK1traction_djumpglobal(getMaterialPropertyByName<RankTwoTensor>("dPK1traction_djumpglobal")),
+    _dPK1traction_dF(getMaterialPropertyByName<RankThreeTensor>("dPK1traction_dF"))
+{
+  if (getParam<bool>("use_displaced_mesh") == true)
+    mooseError("CZMInterfaceKernelIncremental cannot be used with use_displaced_mesh = true. Large "
+               "area changes and rotations are included if the large_kinematic parameter is true "
+               "in the associated cohesive material.");
```

Comment:
you may want to suppress `use_displaced_mesh `. 

params.suppressParameter<bool>("use_displaced_mesh");